application: serial_lte_modem: Add power-off command (#9687)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/Generic_AT_commands.rst

Power saving #XSLEEP
====================

> The ``#XSLEEP`` command makes the nRF9160 enter idle or sleep mode.

Reviewer: ```suggestion
The ``#XSLEEP`` command makes the nRF9160 development kit (DK) enter idle or sleep mode.
```

Reviewer: Would saying that the command makes the modem sleep be correct since we are talking of the Serial LTE modem application?

Author: Not really, modem power on/off is controlled by the app core (AT commands).
What we are talking is nRF9160 SiP sleeps and wake up as a whole.
https://infocenter.nordicsemi.com/topic/ps_nrf9160/chapters/pmu/doc/operationmodes/frontpage.html?cp=2_0_0_4_0

Reviewer: ```suggestion
The ``#XSLEEP`` command makes the nRF9160 System in Package (SiP) enter idle or sleep mode.
```
And elsewhere where I added DK, use the full "nRF9160 SiP" instead.

> The ``#XSLEEP`` command makes the nRF9160 System in Package (SiP) enter idle or sleep mode.

Reviewer: Here the modes are sleep and idle, but in the syntax section they are Sleep and Idle. Which is the correct version?

Also, for the shutdown, it uses a third version: SYSTEM OFF mode. Is there a reason it is capitalized differently?

Author: Sleep and Idle, modes defined by application
System Off, nRF9160 power management mode (see PS)

If you want to do power measurements on the nRF9160 DK while running the SLM application, disable unused peripherals.

Set command
-----------

> The set command makes the nRF9160 enter either idle or sleep mode.

Reviewer: ```suggestion
The set command makes the nRF9160 DK enter either idle or sleep mode.
```

Syntax
~~~~~~

::

   #XSLEEP=<sleep_mode>

The ``<sleep_mode>`` parameter accepts only the following integer values:

* ``0`` - Deprecated.
* ``1`` - Enter Sleep.
  In this mode, both the SLM service and the LTE connection are terminated.

>   The nRF9160 can be waken up using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`.

Reviewer: ```suggestion
  The DK can be woken up using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`.
```

>   The nRF9160 SiP can be waken up using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`.

Reviewer: ```suggestion
  The nRF9160 SiP can be woken up using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`.
```

>   The nRF9160 SiP can be woken up using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`.

Reviewer: ```suggestion

  The nRF9160 SiP can be woken up using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`.
```
Empty line improves readability in the final output.

* ``2`` - Enter Idle.
  In this mode, both the SLM service and the LTE connection are maintained.

>   The nRF9160 can be made to exit idle using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`.

Reviewer: ```suggestion
  The DK can be made to exit idle using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`.
```

>   The nRF9160 SiP can be made to exit idle using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`.

Reviewer: ```suggestion

  The nRF9160 SiP can be made to exit idle using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`.
  
```
Empty line both before and after this sentence to improve readability.

   #XSLEEP: (1,2)
   OK

Power off #XSHUTDOWN
====================

The ``#XSHUTDOWN`` command makes the nRF9160 SiP enter SYSTEM OFF mode, which is the deepest power saving mode.

Set command
-----------

The set command makes the nRF9160 SiP enter SYSTEM OFF mode.

Syntax
~~~~~~

::

   #XSHUTDOWN

.. note::

>    * In this case the nRF9160 SiP cannot be woken up using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`..

Reviewer: ```suggestion
   In this case the nRF9160 SiP cannot be woken up using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`..
```
Don't need a bullet since it's the only thing inside the note.

   #XSLEEP: (1,2)
   OK

Power off #XSHUTDOWN
====================

> The ``#XSHUTDOWN`` command makes the nRF9160 enter SYSTEM OFF mode, which is the deepest power saving mode.

Reviewer: ```suggestion
The ``#XSHUTDOWN`` command makes the nRF9160 DK enter SYSTEM OFF mode, which is the deepest power saving mode.
```

Set command
-----------

> The set command makes the nRF9160 enter SYSTEM OFF mode.

Reviewer: ```suggestion
The set command makes the nRF9160 DK enter SYSTEM OFF mode.
```

Syntax
~~~~~~

::

   #XSHUTDOWN

.. note::

>    * In this mode the nRF9160 cannot wake up using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`..

Reviewer: ```suggestion
   * In this mode the nRF9160 DK cannot wake up using the :ref:`CONFIG_SLM_WAKEUP_PIN <CONFIG_SLM_WAKEUP_PIN>`..
```

Examples
~~~~~~~~

::

   AT#XSHUTDOWN
   OK

Read command
------------

The read command is not supported.

Test command
------------

The test command is not supported.

Reset #XRESET
=============

> The ``#XRESET`` command performs a soft reset of the nRF9160.

Reviewer: ```suggestion
The ``#XRESET`` command performs a soft reset of the nRF9160 DK.
```

Set command
-----------

> The set command resets the nRF9160.

Reviewer: ```suggestion
The set command resets the nRF9160 DK.
```
